src/vault.rs

```diff
@@ -371,26 +371,23 @@ impl VaultSpec {
             &persist_spec.secret_id_path,
             &persist_spec.vault_addr,
         )
-        .await?;
-
-        let raw_read: Result<VaultData, ClientError> =
-            kv2::read(&*client, &persist_spec.kv_mount, &persist_spec.paths.meta).await;
-
-        match raw_read {
-            Ok(value) => {
-                let time_stamp = chrono::DateTime::parse_from_rfc3339(&value.borrow())?;
-                let diff = chrono::Utc::now().signed_duration_since(time_stamp);
-                match diff
-                    > chrono::Duration::milliseconds(
```

Comment:
@srhb ... understand you fixed some issue with "raw_read", but... it looks like - from this diff - that the logic handling should_retry time-diff totally disappeared? Causing "RecentlyTouched" to never be constructed. I would imagine this causing a lot of concurrent issuing attempt. Maybe this explain some of the other problems you are describing?

Reply:
Having #95 could have CI-caught the unused error variant :)